ChronoKeeper/StoryChunkExtractorRDMA.cpp

```diff
@@ -52,7 +46,8 @@ int chronolog::StoryChunkExtractorRDMA::processStoryChunk(StoryChunk*story_chunk
         LOG_DEBUG("[StoryChunkExtractorRDMA] Serialized story chunk size: {}", serialized_story_chunk_size);
 
         std::vector <std::pair <void*, std::size_t>> segments(1);
-        segments[0].first = (void*)(&serialized_buf[0]);
+//        segments[0].first = (void*)(serialized_story_chunk.data());
+        segments[0].first = (void*)(serialized_buf);
         segments[0].second = serialized_story_chunk_size + 1;
```

Comment:
> Segments assignments is only happening in the DEBUG build? Am I missing something?

I don't get it. Segment assignments are not inside any pair of preprocessors? The previous preprocessor ends on line 45 in the new code. The next preprocessor starts on line 54. Maybe you miss line 50 and 51 because they are folded by default?

Reply:
YEs, you are correct1 I've missed the folded endif lines. 
I think that now that we get the correct serialized_buffer_size on line 39,  adding an extra byte on lines 51 and 66 are redundant. I think the code would work without adding this extra byte.

Reply:
Do you mean line 51? It's copy-pasted from Thallium's official example in https://mochi.readthedocs.io/en/latest/thallium/08_rdma.html. It looks like at least they recommend this way.

Reply:
My guess the example sends "\" character to indicate the end of string being transferred . 
I suppose that binary archive gives the correct size of the buffer (line 39).  
I'm sure that the code works with added extra byte on line 51, my question is if it works with line 51 & 66 adjusted not to add this extra byte.  if it doesn't we'd keep the code as is